UltraStar Play/Assets/src/model/song/MutableVoice.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+// this should be internal but tests become impossible
+public class MutableVoice
+{
+    private readonly string m_name;
+    private readonly List<MutableSentence> m_sentences = new List<MutableSentence>();
+
+    public MutableVoice(string name)
+    {
+        if (name == null)
+        {
+            throw new ArgumentNullException("name");
+        }
+        m_name = name;
+    }
+
+    public string GetName()
+    {
+        return m_name;
+    }
+
+    public void AddSentence(MutableSentence sentence)
+    {
+        if (sentence == null)
+        {
+            throw new ArgumentNullException("sentence");
+        }
+        else if (sentence.IsEmpty())
+        {
+            Debug.LogWarning("Ignoring empty sentence (hint: it ends at linebreak: " + sentence.GetLinebreakBeat() + ")");
+        }
+        else
+        {
+            sentence.AssertValidNotes();
+            m_sentences.Add(sentence);
+        }
+    }
+
+    public Voice AsVoice()
+    {
+        AssertValidSentences();
+        List<Sentence> sentences = new List<Sentence>();
+        foreach (MutableSentence ms in m_sentences)
+        {
+            sentences.Add(new Sentence(ms.GetNotes(), ms.GetLinebreakBeat()));
+        }
+        return new Voice(m_name, sentences);
+    }
+
+    private void AssertValidSentences()
+    {
+        // assert that there are any sentences at all, do not silently drop a voice
+        if (m_sentences.Count == 0)
+        {
+            throw new SongBuilderException("Voice '" + m_name + "' has no sentences");
+        }
+        // assert sorted sentences
+        m_sentences.Sort((a, b) => a.GetStartBeat().CompareTo(b.GetStartBeat()));
+        // try to do something sensible with the linebreaks
+        for (int i = 0; i < m_sentences.Count - 1; i++)
```

Comment:
I need to run it at all elements _except_ the last one (note the _-1_). The linebreak 'between' sentences _n_ and _n+1_ is stored in sentence _n_. I'm not sure how to do the "except the last one" in a foreach.
Recursive (alternatively: while?) is probably the way to go here.

(I'm not sure if your comment refers to the Sort operation or the for-loop. Sort is only there because I'm not making any assumptions about my input file, see earlier comment)